### PR TITLE
`pip_parse` and `pip_install` can now parse entry points from wheels

### PR DESCRIPTION
**Pull request by UebelAndre**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Does not include precompiled binaries, eg. `.par` files. See [CONTRIBUTING.md](https://github.com/bazelbuild/rules_python/blob/master/CONTRIBUTING.md) for info
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature (please, look at the "Scope of the project" section in the README.md file)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Currently, there is no way to access binaries (or [entry points](https://packaging.python.org/specifications/entry-points/)) associated with a wheel in python. This has been a rather annoying problem since it expects users to manually generate it themselves. A process for which there's currently there's no guidance on. This PR aims to solve that by generating these targets automatically.

Issue Number: https://github.com/bazelbuild/rules_python/issues/341

## What is the new behavior?

Both [pip_install](https://github.com/bazelbuild/rules_python/blob/9a10fdb84911aad767acb60b603c62108f3228f4/docs/pip.md#pip_install) and [pip_parse](https://github.com/bazelbuild/rules_python/blob/9a10fdb84911aad767acb60b603c62108f3228f4/docs/pip.md#pip_parse) now generate `py_binary` targets for each [entry point](https://packaging.python.org/specifications/entry-points/) in a wheel. These are accessible via the `entry_point` macro which has been added to the `requirements.bzl` files generated by these repository rules.

To use this new functionality, you simply load the `entry_point` macro:
```
load("@pip_deps//:requirements.bzl", "entry_point")

alias(
    name = "flake8",
    actual = entry_point("flake8"),
)
```

Then. the target can be run just like any other `py_binary`.
```
bazel run //:flake8 -- --version
```
```
3.9.2 (mccabe: 0.6.1, pycodestyle: 2.7.0, pyflakes: 2.3.1) CPython 3.9.6 on Darwin
```

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: examples/pip_install/WORKSPACE [hunk 1]
```diff
@@ -36,7 +36,7 @@ pip_install(
     #environment = {"HTTP_PROXY": "http://my.proxy.fun/"},
 
     # Uses the default repository name "pip"
-    requirements = "//:requirements.txt",
+    requirements = "//:requirements.in",
```
Comment (alexeagle) [inline examples/pip_install/WORKSPACE:39]: I'd still rather just lock these dependencies for linux to make CI pass. It's healthier for users to see examples with pinned transitive deps.

Comment (UebelAndre) [reply examples/pip_install/WORKSPACE:39]: Can you suggest a way to do that? The way these examples run in CI is unlike any other rule set but I have no problem making that change

Comment (alexeagle) [reply examples/pip_install/WORKSPACE:39]: just what it says in `requirements.txt`: run `bazel run //:requirements.update` on linux

Comment (UebelAndre) [reply examples/pip_install/WORKSPACE]: Updated.

**Comment by alexeagle** [comment]
This is awesome! thanks, I'll merge and go patch into a client codebase and see if we can remove some of our wrappers.

I bet the working directory issue will still be a problem, but that's a missing feature in `py_binary` to essentially `cd` right before invoking the python interpreter

**Review by alexeagle — APPROVED** [review]
🌮

**Pull request merged**